ports/stm32f4/common-hal/busio/UART.c

```diff
@@ -40,8 +40,6 @@
 
 
 STATIC bool reserved_uart[10];
-//TODO: remove this horrible hack
-STATIC busio_uart_obj_t * context_pointers[10]; //numbered by uart module
 
 void uart_reset(void) {
```

Comment:
I think you need to set all reserved_uart entries to true here first. That way only available uarts are false.

Reply:
I don't follow, why would that be? Reserved UART works the same as all the other busio items - uart entries are zero until inited, where they're reserved.

Reply:
K, I was thinking about the other entries for unsupported UART indices. They shouldn't be read because the pins won't reference them.